README.md

```diff
@@ -1,81 +1,167 @@
+
+```typescript
+// Using new
+let user = new User();
+
+// Using an object in the constructor
+user = new User({
+  firstName: "Homer",
+  lastName: "Simpson",
+  manager: {  // you can simply pass an initializer object for this nested message field
+      firstName: "Montgomery",
+      lastName: "Burns"
+  },
+});
+```
+
+In addition, you can use direct property access and work with the same familiar JavaScript syntax you're used to.  For example:
+
+```typescript
+user.active = true;
+user.manager.firstName = "Monty";
+msg.locations = ["Springfield"];
+msg.projects = {
+   "SPP": "Springfield Power Plant"
+};
+```
+
+
 To learn more, have a look at a complete [code example](https://github.com/bufbuild/protobuf-es/tree/main/packages/protobuf-example), 
 the documentation for the [generated code](https://github.com/bufbuild/protobuf-es/blob/main/docs/generated_code.md), 
 and the documentation for the [runtime API](https://github.com/bufbuild/protobuf-es/blob/main/docs/runtime_api.md).
 
+## Getting Started
 
-### How does this compare to protoc's JavaScript generator?
+Installation instructions vary depending on your purpose.  The instructions for npm are shown below but Yarn and pnpm are also supported.
 
-[`js_generator.cc`](https://github.com/protocolbuffers/protobuf-javascript/blob/main/generator/js_generator.cc)
-is rarely updated, and has fallen behind the quickly moving world of JavaScript.
+### [@bufbuild/protobuf](https://www.npmjs.com/package/@bufbuild/protobuf)
 
-For example:
-- it does not support ECMAScript modules
-- it cannot generate TypeScript (3rd party plugins are necessary)
-- it does not support the [canonical JSON format](https://developers.google.com/protocol-buffers/docs/proto3#json)
-- it does not carry over comments from your `.proto` files
+```bash
+npm install @bufbuild/protobuf
+```
 
-Because of this, we want to provide a solid, modern alternative with Protobuf-ES.
+Provides the runtime library, containing base types, generated well-known types, and core functionality.  This installation is required.
 
-The main differences of the generated code:
-- we use plain properties for fields, where protoc uses getter and setter methods
-- we implement the canonical JSON format
-- we generate [much smaller bundles](packages/protobuf-bench)
-- we rely on standard APIs instead of the [Closure Library](http://googlecode.blogspot.com/2009/11/introducing-closure-tools.html)
+Documentation for getting started with the runtime library can be found [here](packages/protobuf).
 
-See the [migration guides](docs/migrating.md) for details.
+### [@bufbuild/protoc-gen-es](https://www.npmjs.com/package/@bufbuild/protoc-gen-es)
 
+**Note**:  If you are unsure about which installation to run, this is the most likely place to start.
 
-### What features are implemented
+```bash
+npm install @bufbuild/protobuf @bufbuild/protoc-gen-es
```

Comment:
Technically `@bufbuild/protoc-gen-es` should be a dev dependency.
Also are we recommending `@bufbuild/protobuf` as a peer dependency if you're shipping a library?